Added Google Style Python Docstrings to the module

iraquitan opened the pull request:
Added docstrings by following the [Google Style](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). For the docstrings I used the links provided by the README, but some descriptions can be improved.

iraquitan committed:
docs: Add suitable docstrings to the module

Add Google Style Python Docstrings to the module, classes, and functions
of pycnn.py file.

Resolves #27

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -41,16 +41,45 @@
         return dx.reshape(self.m * self.n)
 
     def cnn(self, x):
+        """Approximate sigmoidal kernel function for the neural network."""
```
This function is missing arguments description.
x is input to the piece-wise linear sigmoid function.
Please update the function description as "piece-wise linear sigmoid function".

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -41,16 +41,45 @@
+    """
 
     def f(self, x, t, Ib, Bu, tempA):
+        """Computes the derivative of x at t."""
```
This function is missing arguments description.
Ib is system bias.
Bu is convolution of control template with input.
tempA is feedback template.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -41,16 +41,45 @@
+
+    Attributes:
+        n (int): The number of rows from the image matrix.
+        m (int): The number of columns form the image matrix.
```
Additionally, n is the height and m is the width of the image.

ankitaggarwal011 reviewed (commented):
Please look at the specific comments. Thanks.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -67,6 +96,20 @@ def validate(self, input_location):
+            inputlocation (str): The string path for the input image.
+            outputlocation (str): The string path for the output processed
+                image.
+            tempA (:obj:`list` of :obj:`list`of :obj:`float`): The template
```
Please refer tempA (template A) as the feedback template and tempB (template B) as control template everywhere.

The above comment " # tempA: control template, tempB: feedback template" is wrong, and I'll update it after merging this request. Thanks.

ankitaggarwal011 commented:
@iraquitan, I've added the init method to the object to initialize m and n variables. Would you please document it as well while making the other changes. Thanks a lot. Great work!

Please make sure your working on the updated version of the project.

iraquitan committed:
Merge pull request #1 from ankitaggarwal011/master

Update from original

iraquitan committed:
Merge branch 'issue27'

iraquitan committed:
docs: Update docstrings descriptions.

Update some docstrings descriptions provided by the owner of original
repository. Also add inline docstring to __init__ method.

iraquitan commented:
Added the requested changes.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -41,20 +41,62 @@
         return dx.reshape(self.m * self.n)
 
     def cnn(self, x):
+        """"Piece-wise linear sigmoid function.
```
There is an extra quote here.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
+                template.
+            tempB (:obj:`list` of :obj:`list`of :obj:`float`): Control
+                template.
+            initialcondition (float): The initial template state.
```
initialcondition isn't related to template, you can simply refer it as "The initial state." Please change it at other places as well.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
+            tempB (:obj:`list` of :obj:`list`of :obj:`float`): Control
+                template.
+            initialcondition (float): The initial template state.
+            Ib (float): The template parameter z.
```
Ib is system bias, the parameter z in a cellular neural network. Please change it at other places as well.

ankitaggarwal011 commented on pycnn.py:
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
+                template.
+            initialcondition (float): The initial template state.
+            Ib (float): The template parameter z.
+            t (numpy.ndarray): A numpy array with evenly spaced numbers.
```
"A numpy array with evenly spaced numbers representing time points". Please change it at other places as well.

ankitaggarwal011 commented:
@iraquitan, you've done an excellent job. Kudos! There are only a few minor corrections and we're good to go. Thanks a lot for your contribution. I appreciate your help. :+1:

iraquitan committed:
docs: Update docstring descriptions and typo fix

iraquitan commented:
Added and fixed the requested changes.

ankitaggarwal011 commented:
Excellent work @iraquitan. Thank you for your contribution. Your help is much appreciated. :+1:
